### PR TITLE
Added guide about creating an OWA

### PR DESCRIPTION
**Pull request by deniskomissarov**
We have added a guide for new developers about how to create an Open Web App, as we were told this is the new standard. Here is the talk post about this topic: https://talk.openmrs.org/t/updating-new-module-documentation-to-explain-gsp/16654

The book has been reorganized to have a clearer structure and make it easier to find relevant information. Additionally, pages have been added about JavaScript coding conventions and tips. We have also heard that modules are still used, but primarily for backend work. A short page has been added that briefly describes the differences, and attempts to help people choose whether their project needs a module or an OWA.

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can you squash these commits into one as advised at https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by deniskomissarov** [comment]
As requested, the commits have been squashed into one.

**Comment by dkayiwa** [comment]
@deniskomissarov i still see two commits above.

**Comment by deniskomissarov** [comment]
@dkayiwa My apologies, I was not aware merge commits could be squashed. I have fixed it now.

**Comment by dkayiwa** [comment]
The pull request has too many changes. Did you actually make all of them? Or are some simply formatting changes?

**Comment by deniskomissarov** [comment]
The majority of the changes that the pull request has are renamed files, since we reorganized the structure of the book. We wanted to group things together in a logical manner so that it is easier to find information than when everything is in one directory. Also, images and other assets were renamed to be more consistent, so we had to modify every file where they were used.

Despite this, most of the work actually involved creating a guide on making OWAs and adding information about JavaScript coding conventions. You can find the OWA guide [here](https://github.com/deniskomissarov/openmrs-book-developer-manual/blob/master/en/Development_In_OpenMRS/Getting_Started/CreatingAnOWA.md) and the coding conventions [here](https://github.com/deniskomissarov/openmrs-book-developer-manual/tree/master/en/Development_In_OpenMRS/Coding_Conventions_%26_Tips).

**Comment by dkayiwa** [comment]
Can you do that as separate pull requests?

File: en/About_OpenMRS/OpenMRSToday.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <center><h1>OpenMRS Today</h1></center>
 ********************************************************
 
-<center><img src="/assets/openmrs_atlas_2016.png"></center> _ OpenMRS Atlas, October 2016 _
```
Comment (dkayiwa) [inline en/About_OpenMRS/OpenMRSToday.md:4]: Did you intentionally move this and others from "assests" to "res"?

**Comment by dkayiwa** [comment]
@deniskomissarov did you see the above comments?

**Comment by dkayiwa** [comment]
@deniskomissarov ping me to reopen this when you resume working on it.

**Pull request closed without merging**